controlplane/kubeadm/internal/workload_cluster_coredns.go

```diff
@@ -125,13 +126,18 @@ func (w *Workload) UpdateCoreDNS(ctx context.Context, kcp *controlplanev1.Kubead
 		return err
 	}
```

Comment:
Is there a way to do this without having to "get" the ClusterRole every time we execute UpdateCoreDNS (which is afaik on every KCP reconcile)?

Or, maybe it doesn't matter because we are caching all ClusterRoles currently anyway? (IIRC we don't set ClusterRole as uncached so the Get call should give us an informer & cache currently anyway?)

Reply:
I don't see any other way. And yes, the ClusterRole is cached so I think this is ok.

Reply:
Hm we could skip it if Kubernetes >= v1.23 and CoreDNS FromImageTag >= 1.8.1 I guess?

This would mean that the ClusterRole was patched before. Not sure if it makes the logic too complicated. But it would mean that for most current clusters we would never execute the code and we would never create an informer+cache for the ClusterRole.

(which essentially means going forward this will be never done)

Reply:
👍  to skipping for >= v1.23 .

CoreDNS FromImageTag >= 1.8.1,no because:

* k8s v1.21 + coredns >= 1.8.1 -> no endpointslices
* k8s v1.22 + coredns >= 1.8.1 (keep same coredns version) -> endpointslices

Reply:
Logic is now wway simpler :-)